Automattic/wp-calypso
# Site preview domain upsell: add callout to the iframe preview

#### Proposed Changes

Add the domain upsell callout to the site preview iframe (3 dots + view on posts lists). It should only show for sites on a free plan that don't have a custom domain name.
Domain upsell component is based on [this one](https://github.com/Automattic/wp-calypso/blob/38d4f581b4fc29f4318eb801f37b7a3c257f18a0/client/components/domains/domain-upsell/index.jsx) from [this PR](https://github.com/Automattic/wp-calypso/pull/72847). 

#### Testing Instructions

- With a free site without a custom domain, go to `/posts`
- Clock on the right 3 dots and click on view
- You should see the domain upsell banner.
- Check View track event `calypso_site_preview_domain_upsell_callout_view`
- Check Click track event: `calypso_site_preview_domain_upsell_callout_click`
- Check Dismiss track event: `calypso_site_preview_domain_upsell_callout_dismiss`

To enable the callout after dismiss it, set the `site_preview_domain_upsell_callout-SITEID` calypso preference to false

#### Screenshots
<img width="1884" alt="image" src="https://user-images.githubusercontent.com/402286/216788910-ab5b7e41-fd0c-4683-8e03-a81c02d1b843.png">

#### Pre-merge Checklist




Fixes https://github.com/Automattic/dotcom-forge/issues/1569
